dev-requirements.txt

```diff
@@ -4,3 +4,5 @@ coverage>=6.3,<8
 setuptools_scm>=7.0,<9.0
 tox>=2.5,<4.0
 pytest-cov>=4.0,<5.0
+mypy>=1.9,<2
+build>=1.2,<2

```

Comment:
Do we need these as dev dependencies? They are not used in the CI pipeline either I think. Or are they necessary for the _ typeshed import?

Reply:
I added them because the way I know to make a package that works is `python -m build` or `python -m build --wheel` and because `mypy` provides `stubtest` which you can use to make sure that the .pyi file "works".

But it's probably true that you don't need build and other build processes will now get the same result. And the stubtest thing is kind of a nice-to-have or should maybe be documented if we want folks to do it (in Makefile?)

So I could definitely remove one or both of these if you think that's better.

Reply:
I think it would make sense to run the type checks also in CI, but does not need to be part of this PR